lib/reline/ansi.rb

```diff
@@ -66,6 +67,12 @@ def self.set_default_key_bindings(config, allow_terminfo: true)
     end
   end
 
+  def self.set_bracketed_paste_key_bindings(config)
+    [:emacs, :vi_insert, :vi_command].each do |keymap|
+      config.add_default_key_binding_by_keymap(keymap, START_BRACKETED_PASTE.bytes, :bracketed_paste_start)
```

Comment:
q: Bracketed paste doesn't seem like a keybinding since it's sent from the terminal, not the user. Why is it set here?

Reply:
Bracketed paste start sequence `\e[200~` is very similar to some keys: `kf9=\E[20~`, `knp=\E[6~`, `kpp=\E[5~`.
We can't distinguish them until we read the 5th or 6th bytes. So treating them as same as keybinding is one of the simple way to implement it.
We can think key bindings (processed in class KeyStroke) as a transformation layer of byte stream (includes many kind of escape sequences) to meaningful symbols.

Reply:
Good point. It's similar to the terminal capabilities. Your explanation that it deals with byte streams made sense. Thanks!